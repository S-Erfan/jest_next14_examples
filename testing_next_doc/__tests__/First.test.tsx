import { render, screen } from "@testing-library/react";
import Home from "@/app/page";

describe("Home testing", () => {
  it("test have logo text in Home", () => {
    render(<Home />); //! ARRANGE
    
    // const myElement = screen.getByText("logo");
    const myElement = screen.getByText(/logo/i); //! ACT

    expect(myElement).toBeInTheDocument(); //! ASSERT
  });

  it("in Home h1 should be ", () => {
    render(<Home />);
    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toBeInTheDocument();
  });

  it("take snapshot", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
