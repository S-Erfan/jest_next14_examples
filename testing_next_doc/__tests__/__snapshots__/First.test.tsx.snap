// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home testing take snapshot 1`] = `
<div>
  <header>
    <nav>
      <div>
        Logo
      </div>
      <ul>
        <li>
          <a
            href="/jest"
          >
            Shoping
          </a>
        </li>
        <li>
          <a
            href="login"
          >
            Login
          </a>
        </li>
        <li>
          Contact Us
        </li>
        <li>
          About Us
        </li>
      </ul>
    </nav>
  </header>
  <main>
    <section>
      <h1>
        this is Home
      </h1>
    </section>
    <section>
      section2
    </section>
    <section>
      section3
    </section>
  </main>
  <footer>
    footer
  </footer>
</div>
`;
